lib/rules/id-length.js

```diff
@@ -116,6 +116,9 @@ module.exports = {
                 }
                 return properties && !parent.computed && parent.key.name === node.name;
             },
+            ImportSpecifier(parent, node) {
+                return parent.local === node && parent.imported.range[0] !== parent.local.range[0];
+            },
```

Comment:
I think we should compare by names, so that `import { x as x }` is not reported because it's basically the same as `import { x }`.

Reply:
We could use [`astUtils.getModuleExportName`](https://github.com/eslint/eslint/blob/59cfc0f1b3bbb62260602579f79bd1c36ab5a00f/lib/rules/utils/ast-utils.js#L792) for `parent.imported` so that `import { "x" as x }` also isn't reported.

Reply:
Done!